drivers/rtc/rtc_ll_stm32.c

```diff
@@ -442,6 +428,16 @@ static int rtc_stm32_set_time(const struct device *dev, const struct rtc_time *t
 	/* WeekDay sunday (tm_wday = 0) is not represented by the same value in hardware,
 	 * all the other values are consistent with what is expected by hardware.
 	 */
+
+	LOG_DBG("Setting clock");
+
+	k_spinlock_key_t key = k_spin_lock(&data->lock);
```

Comment:
To avoid functional changes, the following should be used instead:
```suggestion
	k_spinlock_key_t key;
	int err = k_spin_trylock(&data->lock, &key);
	
	if (err < 0) {
	    return err;
	}
```

cc @erwango for whether the functional change is OK or not

Reply:
For reference. These `set_time` functions use `k_spin_lock`:
- rtc_sam0_set_time
- rtc_numaker_set_time
- rtc_mc146818_set_time
- ifx_cat1_rtc_set_time
- ambiq_rtc_set_time
- rtc_rpi_pico_set_time
- ds1307_set_time

`k_spin_trylock` is not used by any rtc driver atm.

Reply:
@erwango any chance you got time to give this a look?

Reply:
I don't think there's a good response here.
I agree you @mathieuchopstm, you're changing the dynamic here, but drivers behaves today differently from the other drivers indeed, so we can't argue about we'll fail on the API contract.
I'm fine with keeping this change and if some applications have trouble with that, let's say that we're moving to the general API consensus.